src/app/content/components/Sidebar.tsx

```diff
@@ -1,59 +1,106 @@
-// tslint:disable:variable-name
 import { HTMLElement } from '@openstax/types/lib.dom';
 import React, { Component, ComponentType } from 'react';
-import { FormattedMessage } from 'react-intl';
 import { connect } from 'react-redux';
 import styled, { css } from 'styled-components';
+import MobileScrollLock from '../../components/MobileScrollLock';
+import { navDesktopHeight, navMobileHeight } from '../../components/NavBar';
 import theme from '../../theme';
 import { AppState } from '../../types';
 import { isArchiveTree } from '../guards';
 import * as selectors from '../selectors';
-import { ArchiveTree, Book, Page } from '../types';
-import { scrollTocSectionIntoView, stripIdVersion } from '../utils';
+import { ArchiveTree, Book, Page, State } from '../types';
+import { scrollTocSectionIntoView } from '../utils/domUtils';
+import { stripIdVersion } from '../utils/idUtils';
+import {
+  bookBannerDesktopHeight,
+  bookBannerMobileHeight,
+  sidebarDesktopWidth,
+  sidebarMobileWidth,
+  sidebarTransitionTime,
+  toolbarDesktopHeight,
+  toolbarMobileHeight
+} from './constants';
 import ContentLink from './ContentLink';
+import SidebarControl from './SidebarControl';
+import { styleWhenSidebarClosed } from './utils/sidebar';
 
-export const sidebarWidth = 33.5;
-const sidebarTransitionTime = 300;
 const sidebarPadding = 1;
 
-// TODO - magic numbers to be replaced in `top`, `height` when ToC gets real styling in openstax/unified#176
-const SidebarBody = styled.div<{isOpen: boolean}>`
-  top: 23rem;
-  height: calc(100vh - 23rem);
+const sidebarClosedStyle = css`
+  overflow-y: hidden;
+  transform: translateX(-${sidebarDesktopWidth}rem);
+  box-shadow: none;
+  background-color: transparent;
+  ${theme.breakpoints.mobile(css`
+    transform: translateX(-${sidebarMobileWidth}rem);
+  `)}
+
+  > * {
+    visibility: hidden;
+    opacity: 0;
+  }
+`;
+
+// tslint:disable-next-line:variable-name
+const SidebarBody = styled.div<{isOpen: State['tocOpen']}>`
+  position: sticky;
+  top: ${bookBannerDesktopHeight}rem;
+  margin-top: -${toolbarDesktopHeight}rem;
   overflow-y: auto;
-  padding: ${sidebarPadding}rem;
-  transition: all ${sidebarTransitionTime}ms;
+  height: calc(100vh - ${navDesktopHeight + bookBannerDesktopHeight}rem);
+  transition:
+    transform ${sidebarTransitionTime}ms,
+    box-shadow ${sidebarTransitionTime}ms,
+    background-color ${sidebarTransitionTime}ms;
   background-color: ${theme.color.neutral.darker};
+  z-index: 3; /* stay above book content and overlay */
+  margin-left: -50vw;
+  padding-left: 50vw;
+  width: calc(50vw + ${sidebarDesktopWidth}rem);
+  min-width: calc(50vw + ${sidebarDesktopWidth}rem);
+  box-shadow: 0.2rem 0 0.2rem 0 rgba(0, 0, 0, 0.1);
+  ${theme.breakpoints.mobile(css`
+    width: calc(50vw + ${sidebarMobileWidth}rem);
+    min-width: calc(50vw + ${sidebarMobileWidth}rem);
+    margin-top: -${toolbarMobileHeight}rem;
```

Comment:
All these negative signs still worry me about maintainability but at least we have unit test screenshots!

Reply:
yea element positioning is a bit crazy in this layout, hopefully we won't have to do any more after this